bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -1,266 +1,238 @@
+        executor.schedule(new Runnable {
+          def run(): Unit =
+
+          // If dequeue is false this means that (a) another task scheduled *after* me ended up
+          // running before me, so their result is as good as mine would have been and we're done;
+          // or (b) we don't care about that program anymore, so we're done.
+            if (dequeue(key, obs.getProgramID)) {
+              // Otherwise construct an obs context, verify that it's bagworthy, and go
+              ObsContext.create(obs).asScalaOpt.filter(_.isEligibleForBags).foreach { ctx =>
+
+                //   do the lookup
+                //   on success {
+                //      if we're in the queue again, it means something changed while this task was
+                //      running, so discard this result and do nothing,
+                //      otherwise update the model
+                //   }
+                //   on failure enqueue again, maybe with a delay depending on the failure
+                val bagsIdMsg = s"BAGS lookup on thread=${Thread.currentThread.getId} for observation=${obs.getObservationID}"
+
+                def lookup[S, T](optExtract: S => Option[T])(worker: (TpeContext, S) => Unit)(results: Try[S]): Unit = {
+                  results match {
+                    case Success(selOpt) =>
+                      // If this observation is once again in the queue, then something changed while this task
+                      // was running, so discard the result.
+                      if (!state.keys(key)) {
+                        LOG.info(s"$bagsIdMsg successful. Results=${optExtract(selOpt) ? "Yes" | "No"}.")
+                        if (ObservationStatus.computeFor(obs) != ObservationStatus.OBSERVED) {
+                          Swing.onEDT {
+                            obs.getProgram.removeCompositeChangeListener(CompositePropertyChangeListener)
+                            obs.getProgram.removeStructureChangeListener(StructurePropertyChangeListener)
+                            worker(TpeContext(obs), selOpt)
+                            obs.getProgram.addStructureChangeListener(StructurePropertyChangeListener)
+                            obs.getProgram.addCompositeChangeListener(CompositePropertyChangeListener)
+                          }
+                        }
+                      }
+
+                    // We don't want to print the stack trace if the host is simply unreachable.
+                    // This is reported only as a GenericError in a CatalogException, unfortunately.
+                    case Failure(CatalogException((e: GenericError) :: _)) =>
+                      LOG.warning(s"$bagsIdMsg failed: ${e.msg}")
+                      enqueue(obs, 5000L)
+
+                    // If we timed out, we don't want to delay.
+                    case Failure(ex: TimeoutException) =>
+                      LOG.warning(s"$bagsIdMsg failed: ${ex.getMessage}")
+                      enqueue(obs, 5000L)
```

Comment:
The comment says "we don't want to delay" but then we seem to delay 5 seconds anyway?